### PR TITLE
Dev-4696 CFDA Table

### PR DESCRIPTION
**Pull request by jonhill13**
**Do Not Merge**
I would rather merge https://github.com/fedspendingtransparency/usaspending-website/pull/1567 and just close this PR to avoid any conflicts.

**High level description:**

Implements the CFDA Table for Financial Assistance Pages.

**Technical details:**

Mainly used the Fed Accounts Section and Workflow.  Adds BaseCFDA. Implements sorting on the frontend.

**JIRA Ticket:**
[DEV-4696](https://federal-spending-transparency.atlassian.net/browse/DEV-4696)

**Mockup:**
https://bahdigital.invisionapp.com/share/57IAC6CHK84#/screens/295998954
https://bahdigital.invisionapp.com/d/main?redirHash=#/console/14220692/296019781/preview?scrollOffset=1040

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR
- [x] Verified cross-browser compatibility
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [x] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension) Confirmed with award ASST_NON_CTA01T340_1450. Local 50 issues and dev there are 70 issues.
- [x] Added Unit Tests for methods in Container Components, reducers, and helper functions
- [x] All componentWillReceiveProps, componentWillMount, and componentWillUpdate in relevant child/parent components/containers replaced with [future compatible life-cycle methods](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) per [JIRA item 3339](https://federal-spending-transparency.atlassian.net/browse/DEV-3339)
- [N/A] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [N/A] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [N/A] Design review complete
- [N/A] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [ ] Code review complete

### CODE REVIEW

File: src/_scss/pages/award/awardFinancialAssistance.scss [hunk 1]
```diff
@@ -23,6 +23,146 @@
...
+                    font-size: rem(20);
+                    line-height: rem(45);
+                  }
+                  .description {
+                    font-size: rem(16);
+                    line-height: rem(10);
+                  }
+                }
+            }
+              .cfda-section__viz {
```
Comment (ebdabbs) [inline src/_scss/pages/award/awardFinancialAssistance.scss]: Could this block be moved to its own file?

File: src/_scss/pages/award/visualizations/_cfdaSectionTable.scss [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      .cfda-section-table__head-cell {
+          @import '../../recipientLanding/header';
+          @import '../../stateLanding/sorter';
+          background-color: $color-white;
+          border: 0;
+          padding: rem(15) rem(2);
+          .header-sorter {
+            .header-sorter__button {
+              svg {
+                fill: #e1e2e3;
```
Comment (ebdabbs) [inline src/_scss/pages/award/visualizations/_cfdaSectionTable.scss]: Prefer to use a variable for a pre-defined color -- this looks close to `$color-gray-warm-light`

File: src/js/app.jsx [hunk 3]
```diff
@@ -39,7 +39,8 @@ import {
     faUserTie,
     faShareAlt,
     faEnvelope,
-    faLink
+    faLink,
+    faArrowCircleLeft
```
Comment (ebdabbs) [inline src/js/app.jsx]: Would like to keep these in alphabetical order if possible

File: src/js/models/v2/awardsV2/BaseCFDA.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    populate(data) {
+        this.samWebsite = data.sam_website || '';
+        this.cfdaWebsite = data.cfda_website || '';
+        this.cfdaFederalAgency = data.cfda_federal_agency || '';
+        this.cfdaNumber = data.cfda_number || '';
+        this.cfdaTitle = data.cfda_title || '--';
+        this.applicantEligibility = data.applicant_eligibility || '';
+        this.beneficiaryEligibility = data.beneficiary_eligibility || '';
+        this.cfdaObjectives = data.cfda_objectives || '';
+        this._totalFundingAmount = isNumber(data.total_funding_amount) ? data.total_funding_amount : '';
```
Comment (ebdabbs) [inline src/js/models/v2/awardsV2/BaseCFDA.js]: It looks like `total_funding_amount` currently represents `federal_action_obligation_amount` + `non_federal_funding_amount` for the individual CFDA, which should not be used for the `percentOfTotal` calculation. We could either add another property to `cfda_info` with the total for all CFDAs associated with the given award, or calculate the total on the frontend.

File: src/js/app.jsx (hunk 3 above)
Comment (jonhill13) [reply src/js/app.jsx]: I could not agree more.

File: src/_scss/pages/award/awardFinancialAssistance.scss (hunk 1 above)
Comment (jonhill13) [reply src/_scss/pages/award/awardFinancialAssistance.scss]: yeah

**Comment by ebdabbs** [comment]
Closing since #1567 includes the latest changes

**Pull request closed without merging**